src/syscalls/interfaces/uhyve.rs

```diff
@@ -215,78 +216,41 @@ impl SyscallInterface for Uhyve {
 	/// ToDo: Add an Errors section - What happens when e.g. malloc fails, how is that handled (currently it isn't)
 	#[cfg(target_os = "hermit")]
 	fn get_application_parameters(&self) -> (i32, *const *const u8, *const *const u8) {
-		//FIXME: Determine how to make this safer, check return values of malloc etc
-		// and then remove the unsafe block and only wrap the parts where it is needed
-		unsafe {
-			// determine the number of arguments and environment variables
-			let mut syscmdsize = SysCmdsize::new();
-			uhyve_send(UHYVE_PORT_CMDSIZE, &mut syscmdsize);
-
-			// create array to receive all arguments
-			let argv_raw = crate::__sys_malloc(
-				syscmdsize.argc as usize * mem::size_of::<*const u8>(),
-				mem::size_of::<*const u8>(),
-			) as *mut *const u8;
-			let argv_phy_raw = crate::__sys_malloc(
-				syscmdsize.argc as usize * mem::size_of::<*const u8>(),
-				mem::size_of::<*const u8>(),
-			) as *mut *const u8;
-			let argv = slice::from_raw_parts_mut(argv_raw, syscmdsize.argc as usize);
-			let argv_phy = slice::from_raw_parts_mut(argv_phy_raw, syscmdsize.argc as usize);
-			for i in 0..syscmdsize.argc as usize {
-				argv[i] = crate::__sys_malloc(
-					syscmdsize.argsz[i] as usize * mem::size_of::<*const u8>(),
-					1,
-				);
-				argv_phy[i] =
-					paging::virtual_to_physical(VirtAddr(argv[i] as u64)).as_u64() as *const u8;
-			}
+		// determine the number of arguments and environment variables
+		let mut syscmdsize = SysCmdsize::new();
+		uhyve_send(UHYVE_PORT_CMDSIZE, &mut syscmdsize);
+
+		// create array to receive all arguments
+		let mut argv = Box::new(vec![ptr::null(); syscmdsize.argc as usize]);
+		let mut argv_phy = vec![ptr::null(); syscmdsize.argc as usize];
+		for i in 0..syscmdsize.argc as usize {
+			argv[i] = crate::__sys_malloc(syscmdsize.argsz[i] as usize * mem::size_of::<u8>(), 1);
+			argv_phy[i] =
+				paging::virtual_to_physical(VirtAddr(argv[i] as u64)).as_u64() as *const u8;
+		}
```

Comment:
How about to create the vectors with `std::Vec::with_capacity instead of filling them first with null pointers and than changing the values?

Reply:
That is true. I fixed it.